architecture-building-systems/CityEnergyAnalyst
# 2349 suspend simulations

This PR solves issue #2349 (feature request: suspend simulations). Actually, the processes running the simulations are terminated, not suspended - due to how scripts work in the CEA.

This branch goes hand in hand with the sister-branch [CityEnergyAnalyst-GUI@2349-suspend-simulations](https://github.com/architecture-building-systems/CityEnergyAnalyst-GUI/tree/2349-suspend-simulations) - you'll need to run that version of the GUI to test this.

To test the new functionality:

- start the GUI
- open project, open scenario
- start a long-running simulation (e.g. `demand`)
- in the JobInfoCard (see image below), chose "Cancel" to cancel the simulation
- check the TaskManager to make sure the processes really are removed

![image](https://user-images.githubusercontent.com/2969564/68761214-e8512a00-0613-11ea-9630-339aa095c8a2.png)

Note that the "Cancel" button is only visible for jobs that are in the "Pending" or "Running" state - jobs that are completed, canceled or threw an error cannot be canceled.
